commons/src/main/java/org/infinispan/commons/dataconversion/MediaTypeResolver.java

```diff
@@ -0,0 +1,57 @@
+package org.infinispan.commons.dataconversion;
+
+import java.io.BufferedInputStream;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+
+import org.infinispan.commons.util.Util;
+
+/**
+ * Resolve media types for files
+ *
+ * @since 10.1
+ */
+public final class MediaTypeResolver {
+
+   private static final Map<String, String> FILE_MAP = new HashMap<>();
+
+   static {
+      InputStream in = null;
+      BufferedInputStream bis = null;
+      try {
+         in = MediaTypeResolver.class.getClassLoader().getResourceAsStream("mime.types");
+         if (in != null) {
+            bis = new BufferedInputStream(in);
+            Scanner scanner = new Scanner(bis);
+            while (scanner.hasNextLine()) {
+               String line = scanner.nextLine();
+               if (!line.startsWith("#")) {
+                  String[] split = line.split("\\s+");
+                  if (split.length > 1) {
+                     String mediaType = split[0];
+                     for (int i = 1; i < split.length; i++) FILE_MAP.put(split[i], mediaType);
+                  }
+               }
+            }
+         }
+      } finally {
+         Util.close(in, bis);
+      }
+   }
+
+   private MediaTypeResolver() {
+   }
+
+   /**
+    * @param fileName The file name
+    * @return The media type based on the internal mime.types file, or null if not found.
+    */
+   public static String getMediaType(String fileName) {
+      if (fileName == null) return null;
```

Comment:
Optional instead of null ?

Reply:
Or empty string or something by default ...

Reply:
I'm not a big fan of Optional and empty string would be an invalid mime type :smile: